frontend/pages/_document.tsx

```diff
@@ -1,9 +1,17 @@
 import React from "react";
-import Document, { Html, Head, Main, NextScript } from "next/document";
+import Document, {
+    Html,
+    Head,
+    Main,
+    NextScript,
+    DocumentContext,
+} from "next/document";
 import { ServerStyleSheet } from "styled-components";
 
+// custom document declaration
```

Comment:
IMO this is not a very necessary comment because it doesn't really describe its purpose, which is to make styled-components work properly